### PR TITLE
feat(465): support notification plugins

### PR DESCRIPTION
**Pull request by d2lam**
Emits an event called `build_status` when a build status updated. 
Notification plugins will subscribe to this event and perform some notification actions.

Should be able to configure multiple notifications like:
```
notification:
   email: ...
   slack: ...
   hipchat: ...
```
Related: https://github.com/screwdriver-cd/screwdriver/issues/465
~Blocked by: https://github.com/screwdriver-cd/data-schema/pull/108~
Blocked by: `screwdriver-notifications-email` to be done so we can test it.

### CODE REVIEW

File: config/default.yaml [hunk 1]
```diff
@@ -143,6 +143,12 @@ bookends:
     teardown:
         - screwdriver-artifact-bookend
 
+notification:
```
Comment (FenrirUnbound) [inline config/default.yaml]: If this is supposed to be a stub setting, I think YAML expects you to initialize it as an empty object `{}`

File: package.json [hunk 2]
```diff
@@ -76,8 +76,8 @@
     "requestretry": "^1.12.0",
     "screwdriver-artifact-bookend": "^1.0.1",
     "screwdriver-build-bookend": "^2.0.1",
-    "screwdriver-config-parser": "^3.1.0",
-    "screwdriver-data-schema": "^16.0.1",
+    "screwdriver-config-parser": "^3.3.0",
```
Comment (FenrirUnbound) [inline package.json]: Until we have a more formalized process, we should at least note these versions in the compatibility document

File: config/default.yaml (hunk 1 above)
Comment (d2lam) [inline config/default.yaml]: @FenrirUnbound  It is set as `{}` in registerPlugins.js. Is that what you meant?

**Comment by d2lam** [comment]
@FenrirUnbound  @petey  Sorry but I updated the PR to send more fields. The change is mainly in `build/update.js`. Could you guys please re-review! :)

File: config/default.yaml (hunk 1 above)
Comment (FenrirUnbound) [inline config/default.yaml]: Kinda like how this is set to an empty object: https://github.com/screwdriver-cd/screwdriver/blob/2af654612965eeaac5719b0cff6c3f1c5393d3a8/config/default.yaml#L104

**Pull request merged**